### DISCUSSION TITLE
Enable extra security open-webui behind reverse proxy on Synology

### DISCUSSION THREAD

**Discussion by stevenbaert** [open-webui/open-webui#4512] [General]
Hi,

Love your product! I have it working in Docker behind a Reverse Proxy on Synology NAS.
That proxy (apparently Nginx underneath) is very basic though.

So I really would like to put my openwebui in the open but with more than user password. I can restrict ip access to my open-webui but that's not feasible for mobile devices.

So I am looking for an easy straight forward solution. Don't like the integration with Google or Microsoft since that oblliges people to sign up there. So any other solution like 2fa (will it be implemented?)  is fine. 

I found your guide https://docs.openwebui.com/tutorial/sso#cloudflare-tunnel-with-cloudflare-access but nowhere a step by step guide.

Could you give some feedback?
Highly appreciated!

Kind regards,
S

**Comment by JPnux** [comment]
hello @stevenbaert 

please can you share your docker run or your docker compose command ?

I failed with open-webui docker installation :-(

I want to use it only with OpenAI API key
I tried this  : 

`docker run -d -p 3000:8080 -e OPENAI_API_KEY=your_secret_key -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main`

**Comment by mpepping** [comment]
#1225 addresses 2FA/MFA support
